index.d.ts

```diff
@@ -0,0 +1,24 @@
+import Vue, { PluginFunction } from 'vue';
+
+export interface VueThinModalOptions {
+  autoMountPortal?: boolean;
+}
+
+declare class VueThinModal {
+  currentName: string;
+  push(name: string): void;
+  pop(): void;
+  replace(name: string): void;
+}
+
+declare class VueThinModalInstaller {
+  static install: PluginFunction<VueThinModalOptions>;
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    readonly $modal: VueThinModal;
+  }
+}
+
+export default VueThinModalInstaller;
```

Comment:
Technically the exported object is not a class. The current fix would allow the users to use it by `new VueThinModal()` which is not an expected usage. It would be more accurate to just declare an object value here.

```ts
declare const _default: {
  install: PluginFunction<VueThinModalOptions>;
};
export default _default;
```

Reply:
@ktsn 🙏 

Squashed again.